### PR TITLE
Define ignore_errors & ignore_transactions

### PR DESCRIPTION
**Pull request by smeubank**
### Background
With improved server side sampling Sentry attempts to already remove some transactions which do not provide value to the user. However this is a known case for many users for a long time. There is the traces sampler function, but the UX for this is bulky when you have a simple use case.

**Use case:** There transcations and errors which a users never wants sent to sentry, to appear in a feed or count against their quota.

**Proposal:** provide simple options which is a list of errors/exceptions and transactions, respectively, which the SDK will then always discard and never send to Sentry

**Preview:** [Event Sampling 
](https://develop-git-smeubank-define-ignore-options.sentry.dev/sdk/features/#event-sampling)
Similar implementations:
https://github.com/getsentry/sentry-php/pull/1503

JS `ignoreErrors`: [Manage Your Error Quota](https://docs.sentry.io/product/accounts/quotas/manage-event-stream-guide/#inboundfilters)

### CODE REVIEW

File: src/docs/sdk/features.mdx [hunk 1]
```diff
@@ -55,6 +55,12 @@ if random() < sample_rate:
     transport.capture_event(event)
 ```
 
+To further simplfiy ignoring certain events from being sent to sentry, it is also sugggested to provide `ignoreTransactions` and `ignoreErrors` (or exception, choose terminology which is best for the platform). This option should provide a simple way to allow users to discard events (ignore) from before they are sent to sentry. Prevents sending data which is is undesired and may consume quota or resources on the Sentry server.
```
Comment (philipphofmann) [inline src/docs/sdk/features.mdx:58]:
```suggestion
To further simplfiy ignoring certain events from being sent to sentry, it is also suggested to provide `ignoreTransactions` and `ignoreErrors` (or exception, choose terminology which is best for the platform). This option should provide a simple way to allow users to discard events (ignore) from before they are sent to sentry. Prevents sending data which is is undesired and may consume quota or resources on the Sentry server.
```

File: src/docs/sdk/features.mdx [hunk 2]
```diff
@@ -55,6 +55,12 @@ if random() < sample_rate:
...
+
+```python
+ignore_transactions = ['GET /api/health','/api/v1/*']
```
Comment (philipphofmann) [inline src/docs/sdk/features.mdx:61]: Please specify that this list refers to transaction names.

**Review by philipphofmann — COMMENTED** [review]
Awesome feature. I added two small notes.

**Comment by marandaneto** [comment]
@smeubank might be worth pointing out to https://develop.sentry.dev/sdk/client-reports/ as well since events are dropped on the client.
We don't have a generic item for that, maybe let's add new ones? should be straightforward.

**Comment by smeubank** [comment]
> @smeubank might be worth pointing out to https://develop.sentry.dev/sdk/client-reports/ as well since events are dropped on the client. We don't have a generic item for that, maybe let's add new ones? should be straightforward.

will merge this as is. @jjbayer is there anything we need to consider on relay to send a new item type client reports for these 2 new discard reasons?

**Pull request merged**